core/src/Kokkos_HostSpace.hpp

```diff
@@ -291,15 +293,17 @@ namespace Kokkos {
 
 namespace Impl {
 
+#define PAR_DEEP_COPY_USE_MEMCPY
```

Comment:
Macros for non-user consumptions should start with KOKKOS_IMPL_
Probably this should be called something like KOKKOS_IMPL_ENABLE_MEMCPY_FOR_DEEP_COPY
or so

Reply:
This should go, it was only for testing.  I don't see a point to providing or maintaining multiple code paths. If memcpy is sucking, we should address that with a vendor. (it doesn't suck anywhere I've seen)

Reply:
The memcpy impl is the default, I've added an OpenMP Target memcpy impl (depends on OpenMP 4.5 API)

I've used your naming convention for the define guarding that